### PR TITLE
Replace 0 and NULL with nullptr.

### PR DESCRIPTION
**Pull request by drwells**
Part of #350.

### CODE REVIEW

File: src/utilities/RNG.cpp [hunk 1]
```diff
@@ -256,7 +256,7 @@ RNG::parallel_seed(int global_seed)
     {
         if (seed == 0)
         {
-            seed = static_cast<int>(std::time(0));
+            seed = static_cast<int>(std::time(nullptr));
```
Comment (boyceg) [inline src/utilities/RNG.cpp]: Looks wrong.

Comment (boyceg) [inline src/utilities/RNG.cpp]: Good catch.

**Comment by boyceg** [comment]
👍

**Comment by drwells** [comment]
I got bear to cooperate by deleting all the 2D build instructions, which caused clang-tidy to miss a bunch of `NULL`s. I am updating this now.

**Pull request merged**